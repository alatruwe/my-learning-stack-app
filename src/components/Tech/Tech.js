import React from "react";

class Tech extends React.Component {
  static defaultProps = {
    updateTech: () => {},
  };
  render() {
    return (
      <button
        onClick={(e) => {
          this.props.updateTech(this.props.tech);
        }}
        type="button"
        className="btn"
      >
        {this.props.tech}
      </button>
    );
  }
}

export default Tech;
