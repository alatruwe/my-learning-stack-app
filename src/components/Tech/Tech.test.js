import React from "react";
import ReactDOM from "react-dom";
import Tech from "./Tech";
import renderer from "react-test-renderer";

describe(`Tch component`, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Tech />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<Tech />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
