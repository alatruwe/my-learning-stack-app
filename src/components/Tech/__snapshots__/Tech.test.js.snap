// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tch component renders the UI as expected 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
/>
`;
